packages/react-conformance/src/defaultErrorMessages.tsx

```diff
@@ -0,0 +1,682 @@
+        resolveErrorMessages([
+          'Make sure that your component can correctly render as' + chalk.hex('#e00000')(' HTML tags') + '.',
+          'Check if you are missing any' +
+            chalk.hex('#e00000')(' requiredProps ') +
+            'within the isConformant in your test file. ',
+          `Make sure that your component's ` +
+            chalk.hex('#e00000')(displayName + '.base.tsx') +
+            ' file contains a valid return statement.',
+          'Check to see if your component works as expected with' +
+            chalk.hex('#e00000')(' mount ') +
+            'and' +
+            chalk.hex('#e00000')(' safeMount') +
+            '.',
+        ]) +
+        receivedErrorMessage(error),
+    );
+    failedTests.push('as-renders-html');
+  },
+
+  'display-failed-tests': (componentInfo: ComponentDoc, testInfo: IsConformantOptions) => {
+    const { displayName } = testInfo;
+    if (failedTests.length > 0) {
+      console.log(
+        paragraph() +
+          chalk.red.bold.underline(displayName) +
+          chalk.white.bold(' failed during the following isConformant tests:') +
+          paragraph() +
+          chalk.white.bold.italic.bgHex('#2e2e2e')(formatObject(failedTests)),
+      );
+    }
+  },
+};
+
+/** Generates the message for resolving the test error.
+ *  @param resolveMessages Why the test is failing.
+ */
+function resolveErrorMessages(resolveMessages: string[]) {
+  const resolveMessage = [];
+
+  for (let i = 0; i < resolveMessages.length; i++) {
+    resolveMessage.push(paragraph() + chalk.cyan(i + 1 + '. ' + resolveMessages[i]));
+  }
+
+  return paragraph() + chalk.yellow.bold('To resolve this issue:') + resolveMessage.join('') + paragraph();
+}
+
+/** Generates the starting default error message: ( "It appears that __displayName__ doesn't have a __errorMessage__." )
+ *  @param testName The conformance test's name.
+ *  @param displayName The component's name.
+ *  @param errorMessage Why the test is failing.
+ */
+function defaultErrorMessage(testName: string, displayName: string, errorMessage: string) {
+  return (
+    paragraph() +
+    chalk.white.bold.italic.bgHex('#2e2e2e')(testName) +
+    paragraph() +
+    chalk.yellow(`It appears that `) +
+    chalk.white(displayName) +
+    chalk.yellow(` doesn't have a ` + errorMessage) +
+    paragraph()
+  );
+}
+
+/** Generates caught error message in defaultTests.
+ *  @param error The caught error message in defaultTests.
+ */
+function receivedErrorMessage(error: string) {
+  return paragraph() + chalk.white.bold.bgRed(`Here's the full error message:`) + paragraph() + error + paragraph(2);
+}
+
+/** Generates a paragraph.
+ *  @param numberOfParagraphs The number of paragraphs to generate.
+ */
+const paragraph = (numberOfParagraphs?: number) => {
+  if (numberOfParagraphs) {
+    const paragraphs = [];
+    for (let i = -1; i < numberOfParagraphs; i++) {
+      paragraphs.push(`
```

Comment:
Just curious, why this approach instead of just creating a series of `os.EOL`? Also it's okay to use `String.prototype.repeat` here since this runs in Node (`repeat` is in ES6, which is supported by all node versions we support).

Reply:
In general for code that runs in Node, we care about supporting Node 8+ (soon to be 10+) which generally aligns with ES2017. That means [everything from ES6](https://kangax.github.io/compat-table/es6/) is allowed, and you can look at [this page for ES2016+](https://kangax.github.io/compat-table/es2016plus/#node8_10).

For code that runs in the browser it's easier to check supported methods--you can just search on https://caniuse.com/ and check the IE 11 column.

Reply:
Thanks for the suggestion! I updated the implementation of the paragraph function.